edx_exams/apps/core/tests/test_api.py

```diff
@@ -289,6 +290,92 @@ def test_with_no_attempt(self):
+            resource_id=str(uuid.uuid4()),
+            course_id=self.course_id,
+            provider=self.test_provider,
+            content_id=self.content_id,
+            exam_name='test_exam',
+            exam_type='proctored',
+            time_limit_mins=30,
+            due_date='2040-07-01 00:00:00',
+            hide_after_due=False,
+            is_active=True
+        )
+
+        self.user = User.objects.create(
+            id=3,
+            username="jerry",
+            email="jerry@example.com",
+            lms_user_id=2
+        )
+
+    def create_mock_exam_attempt(self, user, attempt_number, status, start_time):
+        return ExamAttempt.objects.create(
+            user=user,
+            exam=self.exam,
+            attempt_number=attempt_number,
+            status=status,
+            start_time=start_time,
+            allowed_time_limit_mins=None
+        )
+
+    def test_get_latest_exam_attempt_for_user(self):
+        """
+        Test that the GET function in the ExamAttempt view returns
+        the latest exam attempt for a user
+        """
+
+        one_hour_ago = datetime.now() - timedelta(hours=1)
+        expected_attempt = self.create_mock_exam_attempt(self.user, 1, ExamAttemptStatus.started, datetime.now())
+        self.create_mock_exam_attempt(self.user, 2, ExamAttemptStatus.started, one_hour_ago)
+        self.create_mock_exam_attempt(self.user, 3, ExamAttemptStatus.ready_to_submit, one_hour_ago)
+        self.create_mock_exam_attempt(self.user, 4, ExamAttemptStatus.created, None)
+        self.create_mock_exam_attempt(self.user, 5, ExamAttemptStatus.submitted, None)
+        self.create_mock_exam_attempt(self.user, 6, ExamAttemptStatus.rejected, None)
+        latest_attempt = get_latest_attempt_for_user(self.user.id)
+
+        self.assertEqual(latest_attempt.status, expected_attempt.status)
+        self.assertEqual(latest_attempt.attempt_number, expected_attempt.attempt_number)
+        self.assertEqual(latest_attempt.user.username, expected_attempt.user.username)
+        self.assertEqual(latest_attempt.exam.content_id, expected_attempt.exam.content_id)
+
+    def test_no_attempt_for_user(self):
+        """
+        Test that if the user has no exam attempts, that the endpoint returns None
+        """
+
+        one_hour_ago = datetime.now() - timedelta(hours=1)
+        self.create_mock_exam_attempt(self.user, 1, ExamAttemptStatus.created, datetime.now())
+        self.create_mock_exam_attempt(self.user, 2, ExamAttemptStatus.submitted, one_hour_ago)
+        self.create_mock_exam_attempt(self.user, 3, ExamAttemptStatus.rejected, one_hour_ago)
+
+        self.assertIsNone(get_latest_attempt_for_user(9999999999))
+
+    def test_no_attempts_have_start_time(self):
+        """
+        Test that is the user has no exam attempts with a start time, that the endpoint returns None
+        """
+        self.create_mock_exam_attempt(self.user, 1, ExamAttemptStatus.created, None)
```

Comment:
Similar to above, I think you only need one exam attempt (especially now that the endpoint is not filtering attempts by status).